nupic/research/frameworks/pytorch/imagenet/imagenet_tune.py

```diff
@@ -341,6 +341,33 @@ def run(config):
     address = os.environ.get("REDIS_ADDRESS", config.get("redis_address"))
     ray.init(address=address, local_mode=config.get("local_mode", False))
 
+    # Register serializer and deserializer - needed when logging arrays and tensors.
+    def serializer(obj):
+        if obj.requires_grad:
+            obj = obj.detach()
+        if obj.is_cuda:
+            return obj.cpu().numpy()
```

Comment:
How often and when is this called? 
Coping tensor from GPU to CPU could slow everything down If it is called inside the training or evaluation loop

Reply:
In case it's useful, I've found it best to accumulate logs on the GPU and then copy them to the CPU at the end of the epoch. If you do it that way, there is essentially zero performance cost. Here's example code that I'll submit as a PR soon, after my other PR is merged: https://github.com/mrcslws/nupic.research/blob/907cf9709cc5fd741326baf23a7de5893c7cbcfd/nupic/research/frameworks/pytorch/imagenet/mixins/log_every_loss.py#L65

Reply:
(In case it's not clear, this comment^ is not a suggestion for this PR. It's a guideline for how we can responsibly add more tracing to our code, whether it's via wandb or via the ray result dict.)

Reply:
As mentioned in the last comment. I've only seen this called on the logs. I also only log cpu values, but the `is_cuda` is included just in case.